### PR TITLE
[Gekidou] Adding logic to hide the list of teams on the teams sidebar when needed

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Adding the logic to hide the list of teams whenever you only have one team.

This is going to need extra work to refactor how is shown the server button, because we need to use the existing space with the channels list. The main objective of this PR is introduce the logic needed to extract the info from the store and show it here.

#### Ticket Link
https://community-daily.mattermost.com/boards/workspace/ya4u6bq5hfydznbedstmtyowsy/dbbd497f-f508-4d87-87aa-0299d8245963/7fb750c4-4773-4f99-8aa0-cfa319824150?c=6101f9bd-1ffb-492d-8d9c-17cc891b4a25
https://mattermost.atlassian.net/browse/MM-39702

#### Checklist
- [ ] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [ ] Includes text changes and localization file updates
- [x] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: Android emulator

```release-note
NONE
```

### CODE REVIEW

**Review by enahum — APPROVED** [review]

File: app/components/channel_list/index.tsx [hunk 1]
```diff
@@ -1,20 +1,89 @@
...
+import {makeStyleSheetFromTheme} from '@utils/theme';
 
-import ChannelsList from './channel_list';
+import Categories from './categories';
+import ChannelListHeader from './header';
+import LoadingError from './loading_error';
+import SearchField from './search';
 
-import type {WithDatabaseArgs} from '@typings/database/database';
+// import Loading from '@components/loading';
```
Comment (larkox) [inline app/components/channel_list/index.tsx:17]: Any reason to keep this comment here?

File: app/components/channel_list/index.tsx [hunk 2]
```diff
@@ -1,20 +1,89 @@
...
+        return {maxWidth: withTiming(tabletWidth.value, {duration: 350})};
+    }, []);
+
+    useEffect(() => {
+        if (isTablet) {
+            tabletWidth.value = TABLET_SIDEBAR_WIDTH - (teamsCount > 1 ? TEAM_SIDEBAR_WIDTH : 0);
+        }
+    }, [isTablet, teamsCount]);
+
+    // @to-do; remove after testing
```
Comment (larkox) [inline app/components/channel_list/index.tsx:66]: Should this be removed already?

**Review by larkox — APPROVED** [review]
LGTM just two nits.

File: app/components/channel_list/index.tsx (hunk 1 above)
Comment (enahum) [reply app/components/channel_list/index.tsx:17]: I kept this cause this is for @shazm to remove with his other PR

File: app/components/channel_list/index.tsx (hunk 2 above)
Comment (enahum) [reply app/components/channel_list/index.tsx:66]:
```suggestion
```

**Review by shazm — APPROVED** [review]
I've tested on iOS / Android now and it shows the team list on the left if there are two or more teams, otherwise it's hidden 👍 

There's a slide out animation for leaving a team (but I couldn't see anything if you join one):

https://user-images.githubusercontent.com/456511/144177984-43a75c8c-b835-48e3-8ed3-c6c48feecd4d.mov

@enahum I think you had a comment to see this?

There's some other work around the sidebar which i'm assuming is not part of this PR (the whole feature set is viewable [here](https://mattermost.atlassian.net/wiki/spaces/MSU/pages/1053327503/Multi-Server+Support)) otherwise it looks pretty good to me.

**Comment by enahum** [comment]
> There's some other work around the sidebar which i'm assuming is not part of this PR (the whole feature set is viewable [here](https://mattermost.atlassian.net/wiki/spaces/MSU/pages/1053327503/Multi-Server+Support)) otherwise it looks pretty good to me for just hiding / showing that team list :) 

How is multi server management part of this PR? This is only about hidding the team sidebar

**Pull request merged**